Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp

```diff
@@ -1756,7 +1828,7 @@ auto OMGIRGenerator::addCurrentMemory(ExpressionType& result) -> PartialResult
 {
     static_assert(sizeof(std::declval<Memory*>()->size()) == sizeof(uint64_t), "codegen relies on this size");
 
-    Value* memory = m_currentBlock->appendNew<MemoryValue>(m_proc, Load, Int64, origin(), instanceValue(), safeCast<int32_t>(Instance::offsetOfMemory()));
+    Value* memory = recordHeap(m_currentBlock->appendNew<MemoryValue>(m_proc, Load, Int64, origin(), instanceValue(), safeCast<int32_t>(Instance::offsetOfMemory())), AbstractHeap::InstanceFields);
```

Comment:
It's a little unfortunate that we don't have an FTLOutput equivalent here since we could've forced a heap if we did. I have a feeling someone is going to forget to add an AbstractHeap at some point and it's going to cause problems... Maybe MemoryValue should require a HeapRange in the constructor or we should ASSERT no MemoryValue has a top range?

Reply:
I think adding it to the MemoryValue constructor would only help a little, since the easier thing to forget is actually adding the value to the list of memory values so we can compute the index ranges later. Maybe we can make an `OMGMemoryValue` that takes a reference to the generator and adds itself to the list?